internal/webhooks/patch_validation.go

```diff
@@ -296,7 +296,12 @@ func validateJSONPatchValues(jsonPatch clusterv1.JSONPatch, variableSet map[stri
 					))
 			}
 		} else {
-			if _, ok := variableSet[*jsonPatch.ValueFrom.Variable]; !ok {
+			// Note: We're only validating if the variable name exists without
+			// validating if the whole path is an existing variable.
+			// This could be done by re-using getVariableValue of the json patch
+			// generator but requires a refactoring first.
+			variableName := getVariableName(*jsonPatch.ValueFrom.Variable)
```

Comment:
I'm aware that we're duplicating more logic here, similar to how we did above with the builtin variable validation.

But I think for now it's the pragmatic thing to do. I think long-term we should clean it up so that the webhook can re-use code of the JSON patch generator for validation to allow more powerful validations.